src/inmanta/agent/handler.py

```diff
@@ -483,6 +542,68 @@ def get_client(self) -> protocol.SessionClient:
             self._client = protocol.SessionClient("agent", self._agent.sessionid)
         return self._client
 
+    def get_file(self, hash_id: str) -> Optional[bytes]:
+        """
+        Retrieve a file from the fileserver identified with the given id. The convention is to use the sha1sum of the
```

Comment:
I'm not sure if the second part is really relevant for the `get_file` method. It seems like it would be mostly relevant for the upload.